### PR TITLE
fix(hermit): `deny(unsafe_op_in_unsafe_fn)`

### PR DESCRIPTION
**Pull request by mkroening**
Tracking issue: https://github.com/rust-lang/rust/issues/127747

r? workingjubilee

CC: @stlankes

### CODE REVIEW

File: library/std/src/sys/pal/hermit/os.rs [hunk 1]
```diff
@@ -154,30 +154,29 @@ impl Iterator for Env {
...
+    let mut result = Vec::new();
 
-        for (key, value) in guard.iter() {
-            result.push((key.clone(), value.clone()));
-        }
-
-        return Env { iter: result.into_iter() };
+    for (key, value) in env {
+        result.push((key.clone(), value.clone()));
     }
```
Comment (joboet) [inline library/std/src/sys/pal/hermit/os.rs:163]: I'd use iterator adapters here to save some reallocations:
```suggestion
    let result = env.iter().map(|(key, value)| (key.clone(), value.clone())).collect::<Vec<_>>();
```

File: library/std/src/sys/pal/hermit/os.rs [hunk 2]
```diff
@@ -154,30 +154,29 @@ impl Iterator for Env {
...
+
+    return Env { iter: result.into_iter() };
```
Comment (joboet) [inline library/std/src/sys/pal/hermit/os.rs:165]:
```suggestion
    Env { iter: result.into_iter() }
```

File: library/std/src/sys/pal/hermit/os.rs [hunk 3]
```diff
@@ -154,30 +154,29 @@ impl Iterator for Env {
...
 }
 
 pub fn getenv(k: &OsStr) -> Option<OsString> {
-    unsafe { ENV.as_ref().unwrap().lock().unwrap().get_mut(k).cloned() }
+    ENV.lock().unwrap().as_mut().unwrap().get_mut(k).cloned()
```
Comment (joboet) [inline library/std/src/sys/pal/hermit/os.rs:169]:
```suggestion
    ENV.lock().unwrap().as_ref().unwrap().get(k).cloned()
```

**Review by joboet — APPROVED** [review]
Some small nits, otherwise this looks good to me.

**Comment by joboet** [comment]
r? joboet

**Comment by mkroening** [comment]
Thanks for the swift review!

> Some small nits

All done :)

**Comment by joboet** [comment]
Great!
@bors r+ rollup

**Pull request merged**